documentation/Language.md

```diff
@@ -68,20 +85,44 @@ The type of the variable is then automatically derived. In order to access an el
 
 ### Interfaces
 Like in other programming languages LPG also offers the user to define interfaces and implement them later on. Here is an example how to define an interface. However there is no type for this variable.
-```
+```lpg
 let make-percent = interface
-    percent(): integer(0, 100)
+    percent(): int(0, 100)
 ```
+
 And this is how you implement this:
-**TO DO**
+```lpg
+let printable = interface
+    print(): string-ref
 
+impl printable for string-ref
+    print(): string-ref
+        self
+```
+3
 And this is how you use it:
+```lpg
+let print-percent = (something: make-percent)
+    integer-to-string(something.percent())
 ```
-let print-percent(something: make-percent)
-    integer-to-string(make-percent.percent)
+
+### Structures
+Structures provide the tool to define your own data structures. Like so:
+```lpg
+let t : type = struct
+    a: boolean
+let t-instance : t = t{boolean.true}
```

Comment:
Please omit redundant types from declarations.

Reply:
Those are there for clarification that you can so something like that. Because you can't do that with functions.